components/scream/src/physics/shoc/tests/shoc_linearinterp_tests.cpp

```diff
@@ -0,0 +1,242 @@
+#include "catch2/catch.hpp"
+
+//#include "share/scream_types.hpp"
+#include <algorithm>
+#include <array>
+#include <random>
+#include <thread>
+
+#include "ekat/scream_kokkos.hpp"
+#include "ekat/scream_pack.hpp"
+#include "ekat/scream_types.hpp"
+#include "ekat/util/scream_arch.hpp"
+#include "ekat/util/scream_kokkos_utils.hpp"
+#include "ekat/util/scream_utils.hpp"
+#include "physics/share/physics_constants.hpp"
+#include "physics/shoc/shoc_functions.hpp"
+#include "physics/shoc/shoc_functions_f90.hpp"
+#include "shoc_unit_tests_common.hpp"
+
+namespace scream {
+namespace shoc {
+namespace unit_test {
+
+template <typename D>
+struct UnitWrap::UnitTest<D>::TestShocLinearInt {
+
+  static void run_property()
+  {
+    static constexpr Int shcol    = 2;
+    static constexpr Int km1     = 5;
+    static constexpr auto km2   = km1 + 1;
+    
+    // TEST ONE 
+    // Test interpolation going from midpoint grid to interface grid.  Note that
+    //  in this case the nlev grid is denoted by variable km1 and the nlevi 
+    //  grid is represented by variable km2.  This is because the linear
+    //  interp routine can interpolate from midpoint to interface grid or
+    //  vice versa so notation should be flexible.   
+    
+    // Note that in this test we are testing TWO profils.  The first column
+    //  will be loaded up with potential temperature values while the second
+    //  column will be loaded up with meridional wind values.  
+
+    // Define the interface height grid [m]
+    static constexpr Real zi_grid[km2] = {12500., 7500., 3000., 750., 250.0, 0.};
```

Comment:
Not sure if it matters for this test or not but in general if a real number is a whole number (e.g. 1.0), we can simply write it as an integer (e.g. 1). C/C++ will promote that number to single or double precision correctly. For numbers which are not whole (e.g. 0.5), we can specify them with a decimal as we would normally do. Therefore you can change the above numbers like:
`static constexpr Real zi_grid[km2] = {12500, 7500, 3000, 750, 250, 0};`
For more info: Please see points 1 and 2 on this page: https://acme-climate.atlassian.net/wiki/spaces/NGDNA/pages/1310753260/Porting+Tips+and+Tricks

Same is the case with rest of the declarations below.